src/protocol/boolean/mod.rs

```diff
@@ -0,0 +1,40 @@
+mod prefix_or;
+
+/// A step generator for bitwise secure operations.
+///
+/// For each record, we decompose a value into bits (i.e. credits in the
+/// Attribution protocol), and execute some binary operations like OR'ing each
+/// bit. For each bitwise secure computation, we need to "narrow" the context
+/// with a new step to make sure we are using an unique PRSS.
+///
+/// This is a temporary solution for narrowing contexts until the infra is
+/// updated with a new step scheme.
+struct BitOpStep {
+    count: u8,
+    id: String,
```

Comment:
I don’t think this needs to actually instantiate a string. I think the other ones all avoid this.